File: src/main/java/peregin/tov/gui/MigPanel.java
package peregin.tov.gui;

import net.miginfocom.swing.MigLayout;

import javax.swing.*;

public class MigPanel extends JPanel {

    public MigPanel(String layoutConstraints, String colConstraints, String rowConstraints) {
        super(new MigLayout(layoutConstraints, colConstraints, rowConstraints));
    }
}

File: src/main/scala/peregin/gpv/gui/MigPanel.scala
package peregin.gpv.gui

import scala.swing.{Component, Panel}
import net.miginfocom.swing.MigLayout
import javax.swing.JComponent


class MigPanel(layoutConstraints: String, colConstraints: String, rowConstraints: String) extends Panel {
  peer.setLayout(new MigLayout(layoutConstraints, colConstraints, rowConstraints))

  def add(comp: Component, constraint: String): Unit = {
    peer.add(comp.peer, constraint)
  }

  implicit def swing2scala(c: JComponent): Component = Component.wrap(c)
}
